### PR TITLE
Use naptan api #183

### PR DESCRIPTION
**Pull request by Antonio-John**
-This tickets look to retrieve stops data from the [NaPTAN wesbite.](https://naptan.api.dft.gov.uk/swagger/index.html) and save in the data folder, then as a feather.
-Downloads data and saves as feather if the local copy of stops data is more then 28 days old.

### CODE REVIEW

File: SDG_11.2.1_main.py [hunk 1]
```diff
@@ -32,18 +32,19 @@
...
-# Define the columns wanted from Naptan
-COLS = list(config["NAPTAN_TYPES"].keys())
-NAPTAN_DTYPES = config["NAPTAN_TYPES"]
 # Get the pandas dataframe for the stops data
-stops_df = di.any_to_pd(file_nm="Stops",
-                        zip_link=NAPT_ZIP_LINK,
-                        ext_order=EXT_ORDER,
-                        dtypes=NAPTAN_DTYPES)
-
+today = datetime.now().strftime('%Y%m%d')
```
Comment (jwestw) [inline SDG_11.2.1_main.py:37]: Where is this variable used?

If it's used in the data ingest file should it be defined there?

File: data_ingest.py [hunk 2]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
 
     absolute_path=os.path.join(dir,csv_file)
```
Comment (jwestw) [inline data_ingest.py:396]: Not sure who coded this (possibly me) but just noticing some pep8 compliance stuff here. 

`absolute_path=os.path.join(dir,csv_file)`

should be 

`absolute_path = os.path.join(dir, csv_file)

File: data_ingest.py [hunk 3]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
...
 
-    return absolute_path
\ No newline at end of file
+    return absolute_path
+
+def _get_stops_from_api(url,file_name):
```
Comment (jwestw) [inline data_ingest.py:400]: This looks great!

File: data_ingest.py [hunk 4]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
...
+
+    Returns:
+        None - just saves into the data folder.
+    """
+    # requests page
+    r = requests.get(url)
+
+    # gets content and then writes to csv
+    url_content = r.content
+    csv_file = open(file_name, 'wb')
```
Comment (jwestw) [inline data_ingest.py:411]: any reason you didn't use `with open` here?

File: data_ingest.py [hunk 5]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
...
+    csv_file.write(url_content)
+    csv_file.close()
+
+    return csv_file
```
Comment (jwestw) [inline data_ingest.py:415]: Should this return anything? Contradicts your docstring

File: data_ingest.py [hunk 6]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
...
+
+def _get_latest_stop_file_date(dir):
+    """Gets the latest stop dataset 
+    Returns:
+        None - just saves into the data folder.
+    """
+    # get's a list of files from the directory
+    # files are in the format stops_YYYYMMDD
+    # attempts to extracts all the dates
+    file_list=os.listdir(dir)
```
Comment (jwestw) [inline data_ingest.py:425]: `file_list=os.listdir(dir)`

should be

`file_list = os.listdir(dir)`

Look into using a linter in VS code to automatically do this stuff for you. If you want to go further, look into pre-commits. 

There is other pep-8 stuff which I won't comment on. Just try running a linter :)

File: data_ingest.py [hunk 7]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
...
+    csv_file.close()
+
+    return csv_file
+
+def _get_latest_stop_file_date(dir):
+    """Gets the latest stop dataset 
+    Returns:
+        None - just saves into the data folder.
+    """
+    # get's a list of files from the directory
```
Comment (jwestw) [inline data_ingest.py:422]: Maybe spread out these comments so they are above the code they are describing

File: data_ingest.py [hunk 8]
```diff
@@ -394,4 +395,93 @@ def get_oa_la_file_name(dir):
...
+    # get output path
+    output_path=os.path.join(os.getcwd(),
+                            "data",
+                            "Stops.feather")
+    # output to feather
+    file.to_feather(output_path)
+
+
+
+def get_stops_file(url, file_name, dir):
```
Comment (jwestw) [inline data_ingest.py:457]: Very nice entry point function.

**Review by jwestw — CHANGES_REQUESTED** [review]
Please take a look at the comments and action what you want there. Please also run a linter.

File: SDG_11.2.1_main.py (hunk 1 above)
Comment (Antonio-John) [reply SDG_11.2.1_main.py:38]: Have moved this into the data ingest file

**Comment by Antonio-John** [comment]
I have rebase with master and think this is ready to merge. Please let me know if you spot any further problems :) @jwestw

File: data_ingest.py (hunk 4 above)
Comment (Antonio-John) [reply data_ingest.py:422]: Not sure, i think the only difference is I have to manually close the file. Some more info here https://www.pythonforbeginners.com/files/with-statement-in-python

**Review by jwestw — APPROVED** [review]

**Comment by jwestw** [comment]
Hi @Antonio-John I am getting a file not found error because I don't have ".../SDG_11.2.1/data/stops"

Could you give me some instructions for testing here? Thanks

**Comment by Antonio-John** [comment]
> Hi @Antonio-John I am getting a file not found error because I don't have ".../SDG_11.2.1/data/stops"
> 
> Could you give me some instructions for testing here? Thanks

Thanks for reviewing/Running.

I have uploaded to the FTP server the stops directory with an initial stops data file. I kind of assumed that the FTP transfer server would be working within the pipeline here for others. 

Are you able to run that code to get the directory/initial file or transfer manually?

**Comment by jwestw** [comment]
Looking good.

**Pull request merged**